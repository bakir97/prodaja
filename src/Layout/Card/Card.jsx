import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import Style from "./style";
const styles = Style;

export const MyCard = props => {
  const { classes, naslov, cijena, slika, tezina, id, history } = props;
  return (
    <Card className={classes.card}>
      <CardActionArea id="area" onClick={() => history.push(`/Proizvod/${id}`)}>
        <CardMedia
          component="img"
          alt="slika"
          className={classes.media}
          height="250"
          image={slika}
          title="slika"
        />
        <CardContent classes={{ root: classes.naslov }}>
          <Typography
            noWrap
            className={classes.content}
            gutterBottom
            variant="h5"
            component="h2"
          >
            {naslov}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Typography className={classes.button}>{tezina} KG</Typography>
        <Typography className={classes.button} style={{ marginLeft: "auto" }}>
          {cijena} KM
        </Typography>
      </CardActions>
    </Card>
  );
};
export default withStyles(styles)(MyCard);
