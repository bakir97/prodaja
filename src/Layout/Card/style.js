export default {
  card: {
    width: "100%",
    maxWidth: 400,
    margin: "1rem",
    marginTop: 0
  },
  media: {
    // ⚠️ object-fit is not supported by IE11.
    objectFit: "cover"
  },
  button: {
    color: "transparent",
    background: "linear-gradient(to right, #373b44, #4286f4)",
    "-webkit-background-clip": "text",
    fontFamily: "Kalam",
    fontSize: "1.3rem"
  },
  content: {
    fontFamily: "Kalam"
  },
  naslov: {
    paddingBottom: 0,
    textAlign: "center"
  },
  icona: {
    fontSize: "2rem"
  },
  iconButton: {
    padding: 5
  }
};
