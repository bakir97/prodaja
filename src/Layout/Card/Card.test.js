import { MyCard } from "./Card";
describe("Card", () => {
  const mock = { history: { push: jest.fn() }, id: 1, classes: {} };
  const wrapper = shallow(<MyCard {...mock} />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should call push", () => {
    wrapper.find("#area").simulate("click");
    expect(mock.history.push).toBeCalledWith("/Proizvod/1");
  });
});
