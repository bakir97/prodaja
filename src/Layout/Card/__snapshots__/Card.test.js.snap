// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `
<WithStyles(Card)>
  <WithStyles(CardActionArea)
    id="area"
    onClick={[Function]}
  >
    <WithStyles(CardMedia)
      alt="slika"
      component="img"
      height="250"
      title="slika"
    />
    <WithStyles(CardContent)
      classes={
        Object {
          "root": undefined,
        }
      }
    >
      <WithStyles(Typography)
        component="h2"
        gutterBottom={true}
        noWrap={true}
        variant="h5"
      />
    </WithStyles(CardContent)>
  </WithStyles(CardActionArea)>
  <WithStyles(CardActions)>
    <WithStyles(Typography)>
       KG
    </WithStyles(Typography)>
    <WithStyles(Typography)
      style={
        Object {
          "marginLeft": "auto",
        }
      }
    >
       KM
    </WithStyles(Typography)>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;
